version.mk

```diff
@@ -17,6 +17,23 @@ func init() { Gitref = \"$(GITREF)\"}  "
 # setver updates version.go and gitref.go with VERSION and GITREF vars
 #
 .PHONY:setver
-setver:
+setver: helm-version
 	@printf $(VERSION_GO) | gofmt > version.go
 	@printf $(GITREF_GO) | gofmt > gitref.go
+
+# helm-version automatically updates the versions of Helm charts to match the version set in the Makefile,
+# so that chart versions are also kept in sync when the Teleport version is updated for a release.
+# If the version contains '-dev' (as it does on the master branch, or for development builds) then we get the latest
+# published major version number by parsing a sorted list of git tags instead, to make deploying the chart from master
+# work as expected. Version numbers are quoted as a string because Helm otherwise treats dotted decimals as floats.
```

Comment:
I'm not sure I understand this part.
If a branch is `7.0.0-dev`, shouldn't the helm chart use `7.0.0`?

Reply:
`7.0.0-dev` is what `master` is currently set to (when our most recent release is `6.1.3`)

When we release `7.0.0`, `master` will get updated to `8.0.0-dev`, so this always needs to calculate `n-1`.

Reply:
But don't we publish helm charts from release branches, where the version is correct?
Publishing `master` chart as the previous version could cause issues, e.g. overwriting the released version of the chart

Reply:
No, the Helm charts are published nightly from `master` by a cron job in Drone.

You make a good point though, I think that if we want to tie Helm chart version to Teleport version (as per #5582) then we need to change Drone to make Helm chart releases part of the current tag/promote process instead, so that we publish a correctly versioned copy on the correct branch whenever a release is made. We're trying to push people to install from `https://charts.releases.teleport.dev` rather than cloning `master` (or a tag/branch) so this makes most sense.

I've been resistant to this because it means that pushing a simple Helm chart bug fix is going to require releasing a whole new version of Teleport, which is a relatively heavyweight operation. I don't see a way around this if we want to inextricably tie chart version to Teleport version, though. This is really what the `appVersion` field is supposed to be for - i.e. you version your Helm chart completely separately to your application and then you can interchangeably use different versions of each. I think @klizhentas wanted to version them the same to reduce confusion, but it will make our release process a bit worse.

Reply:
Can we use the "Promote" function of Drone to trigger a Helm chart deployment, without triggering a full release?

Reply:
Yes, we could probably do that. The only question is what we'd be promoting - i.e. normally with Teleport releases, we promote a `tag` build which has `DRONE_TAG` set. If we want to allow promotion of other arbitrary commits/pushes then we'd have to work based on commit hash. We'd likely have to add some logic which refuses to release if the commit hash being promoted is on the `master` branch rather than a release branch, for example.

Reply:
Raised #6887 to track this